Build.c

```diff
@@ -188,6 +189,7 @@ int Build(struct mddev_ident *ident, struct mddev_dev *devlist, struct shape *s,
 		pr_err("array %s built and started.\n", chosen_name);
 	wait_for(chosen_name, mdfd);
 	close(mdfd);
+	close(bitmap_fd);
```

Comment:
this close call, should be executed at the end of `if (s->bitmap_file) {` block, as it is not used outside of it.
also declaration should be moved to "if" block 
`int bitmap_fd = open(s->bitmap_file, O_RDWR);`